### ISSUE TITLE
Convert license to GPLv3 and include by default in Ansible 

### ISSUE THREAD

**Issue by abenokraitis** [CiscoUcs/imc-ansible#12]
In order for joint customers to fully benefit from these modules, having them built-into the Ansible distribution is ideal. Unfortunately, this requires modules being of type license GPLv3. This would be a good idea since most all Cisco networking modules are GPLv3 and are already included in the base Ansible release by default, or are in progress.

Some more info on submitting modules to Ansible:
https://docs.ansible.com/ansible/dev_guide/developing_modules_in_groups.html

**Comment by vvb** [comment]
@abenokraitis Yes, @dsoper2 is working out the details from our end. It looks like that the nexus ansible modules from Cisco are posted as GPLv3 in the ansible repo. We are proceeding with caution on this. The major concern being internal code-reuse into other products after GPLV3'ing it to ansible. For explanation sake, if we contribute 10 modules to the ansible repo as GPLv3, and then add a 11th one as APACHE 2.0 (which we don't want externally shared). Can we  package these 11 modules as APACHE 2.0 into a cisco product offering? I just want to be sure that it does not put any restrictions of how we can internally re-use these modules. Would really appreciate some thoughts on this.

/ping @dsoper2

**Comment by abenokraitis** [comment]
Correct, the IOS, IOS-XR, and NX-OS networking modules are GPLv3 and are included by default in Ansible today. It would be great to have the full suite of Cisco product Ansible modules all included for a seamless customer experience. 

I can only state that the requirement for distributing Ansible modules as part of Ansible is GPLv3 (the open source project itself is licensed this way). I know there are other teams within Cisco evaluating moving to GPLv3 for other Ansible modules and projects and can connect you with them (as they are involving Cisco Legal). Thanks very much!
